### DISCUSSION TITLE
How do I include parameters in markdown text?

### DISCUSSION THREAD

**Discussion by adrfantini** [quarto-dev/quarto-cli#5941] [Q&A] [answered]
### Description

I am using Quarto with Python and Jupyter.  
I need to programmatically generate documents in which the markdown text body and the titles, as well as some computation, depend on parameters.

For example, define a parameter as such:
```
#| tags: [parameters]

color = 'yellow'
```

Then use it in the text:
```
# The case of color {{< param color >}}
{{< color >}} is a nice color.
...
```

But also in computations:
```
# some python code that uses {{< param color >}}
```

How can I do that, without having to have variables and parameters with the same name?

## Resources found:
- https://stackoverflow.com/questions/18878083/can-i-use-variables-on-an-ipython-notebook-markup-cell/47310524#47310524
- https://github.com/ipython/ipython/issues/2958

**Comment by mcanouil** [comment, marked as answer]
You can't in Jupyter.
But you can using `engine: knitr` and the `reticulate` R package, because parameters are set in the yaml frontmatter, thus allowing you to use `{{< meta params.whatever >}}`.

See the following issues/discussions:

- #5401
- #4688
- #3570
- #1520

**Reply by adrfantini** [reply]
Thanks for that, it really helps.

At this point then however I'm not really sure why I should use Quarto in place of RMarkdown, if I need to have the R toolkit as well anyway to be able to do something that basic.

Anyway I'll try to play around with it some more, thank you again!

**Reply by mcanouil** [reply]
The issue is that for R, knitr supports it from the start while Jupyter does not support centralised parameters or inline code.
It's in the roadmap to fill that void in Jupyter within Quarto.
